pkg/openapi/validation.go

```diff
@@ -239,8 +240,45 @@ func (o *Controller) getGVKByDefinitionName(definitionName string) (gvk string,
 	return "", preferredGVK, fmt.Errorf("gvk not found by the given definition name %s, %v", definitionName, versionsTyped.gvks)
 }
 
+func parseGVK(str string) (group, apiVersion, kind string) {
+	if strings.Count(str, "/") == 0 {
+		return "", "", str
+	}
+	splitString := strings.Split(str, "/")
+	if strings.Count(str, "/") == 1 {
+		return "", splitString[0], splitString[1]
+	}
+	return splitString[0], splitString[1], splitString[2]
+}
+
+func groupMatches(gvkMap map[string]bool, group, kind string) bool {
+	if group == "" {
+		ok := gvkMap["core"]
+		if ok {
+			return true
+		}
+	} else {
+		elements := strings.Split(group, ".")
+		for _, item := range elements {
+			if item != "io" {
+				ok := gvkMap[item]
+				if ok {
+					return true
+				} else {
+					if kind == "Ingress" {
+						return false
+
```

Comment:
What is the case it addresses?

Reply:
(https://github.com/kyverno/kyverno/blob/8a0d465d9084963f5d84603633215bbb82f8a645/pkg/openapi/validation_test.go#L146)
These test cases are getting failed if we have not added.

Reply:
What was failed exactly?

Reply:
Previous code we are using. now I removed.